open-craft/frontend-app-course-authoring
# [Temporary] feat: Taxonomy export menu

**Changes added:**

- ExportModal.
- Export menu added. **Note:** Is hidden for system-defined, because the export API doesn't allow export system-defined.
- Functions to make a call to the export API.
- Flow to export a taxonomy.
- System-defined tooltip.

**Screenshots**

![image](https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/f6818337-bef7-41d7-9f0b-8c890e664f99)
![image](https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/223de2c2-4595-4e1d-b6a8-a59a18b58371)

**Testing instructions**

- Verify the tooltip of the badge of a System-defined taxonomy.
- Create a new Taxonomy on the CMS python shell or use the [taxonomy-sample-data](https://github.com/open-craft/taxonomy-sample-data/tree/main).
- Go to the Taxonomy List page, you should be able to see the new taxonomy.
- Click on the tree dots on the new taxonomy card and on `Export`.
- Check the new Export modal.
- Click on `Export` on the Export modal with CSV format.
- Verify the downloaded file.
- Click on `Export` on the Export modal with JSON format.
- Verify the downloaded file.

**Other info**

- Internal ticket: [FAL-3528](https://tasks.opencraft.com/browse/FAL-3528).
- https://github.com/openedx/modular-learning/issues/105
